pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -71,18 +90,38 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctr
 		return res, err
 	}
 
-	return ctrl.Result{}, q.runListRestorePoints(ctx, req)
+	dispatcher := q.makeDispatcher(q.Log, q.Vdb, q.PRunner, q.Password)
+	// Create a TLS secret for the NMA service
+	err := q.generateCertsForNMA(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if e := q.PFacts.Collect(ctx, q.Vdb); e != nil {
+		return ctrl.Result{}, e
+	}
+	pf, found := q.PFacts.findRunningPod()
+	if !found {
+		q.Log.Info("No pods running")
+		return ctrl.Result{}, nil
+	}
+	hostList := q.PFacts.findExpectedNodeIps()
+
+	opts := []restorepointsquery.Option{}
+	// extract out the communal and config information to pass down to the vclusterops API.
+	opts = append(opts,
+		restorepointsquery.WithInitiator(pf.name, pf.podIP),
+		restorepointsquery.WithHosts(hostList),
```

Comment:
Why do we need a split between initiator and host? Can we just provide one  that acts as the initiator and host list. We only need to talk to one host to get this information.